src/main/java/org/gradle/profiler/CliInvoker.java

```diff
@@ -49,15 +55,21 @@ public void runTasks(List<String> tasks, List<String> gradleArgs, List<String> j
 
         ProcessBuilder builder = new ProcessBuilder(commandLine);
         builder.directory(projectDir);
-        if (!daemon) {
+        if (daemon) {
+            builder.environment().put("GRADLE_OPTS", "-Xmx128m -Xms128m -XX:+HeapDumpOnOutOfMemoryError");
```

Comment:
Why add these `GRADLE_OPTS` now?

Reply:
Same as here: https://github.com/gradle/gradle/commit/f3c5ee6185c19b1e2af44fd8ee90b1c9c8338d26

When the build generates too much log, the log events might congest in client JVM and OOM. gradle-profiler doesn't provide an API to configure the client JVM or environment variables, right?

Reply:
Do we already know we need this? If we don't, I wouldn't add it now.

If we need something like that, I would make it configurable via something like `clientJvmArgs` on `CliInvoker`.

Reply:
Yes, we need it for build scan performance test. This is just a workaround, I opened a new issue https://github.com/gradle/gradle-profiler/issues/260 to handle this issue and no-daemon jfr recording issue.